x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/MvExpandOperator.java

```diff
@@ -60,18 +61,30 @@ protected Page process(Page page) {
             noops++;
             return page;
         }
-        if (page.getBlockCount() == 1) {
-            assert channel == 0;
-            return new Page(expandedBlock);
-        }
-
-        int[] duplicateFilter = buildDuplicateExpandingFilter(expandingBlock, expandedBlock.getPositionCount());
+        try {
+            if (page.getBlockCount() == 1) {
+                assert channel == 0;
+                return new Page(expandedBlock);
+            }
 
-        Block[] result = new Block[page.getBlockCount()];
-        for (int b = 0; b < result.length; b++) {
-            result[b] = b == channel ? expandedBlock : page.getBlock(b).filter(duplicateFilter);
+            int[] duplicateFilter = buildDuplicateExpandingFilter(expandingBlock, expandedBlock.getPositionCount());
+
+            Block[] result = new Block[page.getBlockCount()];
+            boolean success = false;
+            try {
+                for (int b = 0; b < result.length; b++) {
+                    result[b] = b == channel ? expandedBlock : page.getBlock(b).filter(duplicateFilter);
```

Comment:
Should we assign the `expandedBlock` to the `result` array before the loop to ensure that we always release it in case we hit the breaker with `b < channel` ?

Reply:
++